erpnext/payroll/doctype/salary_slip/salary_slip.py

```diff
@@ -307,28 +307,59 @@ def get_working_days_details(self, joining_date=None, relieving_date=None, lwp=N
 			if payroll_based_on == "Attendance":
 				self.payment_days -= flt(absent)
 
-			unmarked_days = self.get_unmarked_days()
 			consider_unmarked_attendance_as = frappe.db.get_value("Payroll Settings", None, "consider_unmarked_attendance_as") or "Present"
 
 			if payroll_based_on == "Attendance" and consider_unmarked_attendance_as =="Absent":
+				unmarked_days = self.get_unmarked_days(include_holidays_in_total_working_days)
 				self.absent_days += unmarked_days #will be treated as absent
 				self.payment_days -= unmarked_days
-				if include_holidays_in_total_working_days:
-					for holiday in holidays:
-						if not frappe.db.exists("Attendance", {"employee": self.employee, "attendance_date": holiday, "docstatus": 1 }):
-							self.payment_days += 1
```

Comment:
Removed this since it's already taken care of in:

https://github.com/frappe/erpnext/blob/385830bb51997fb2411a9ac0945a812b2bb47683/erpnext/payroll/doctype/salary_slip/salary_slip.py#L278-L281